src/foam/nanos/crunch/ui/UCJView.js

```diff
@@ -183,28 +182,35 @@ foam.CLASS({
     },
 
     async function onSave(isValid, ucj) {
-      if ( isValid && ucj.status != this.CapabilityJunctionStatus.ACTION_REQUIRED ) {
-        this.notify(this.SUCCESS_UPDATED, '', this.LogLevel.INFO, true);
-        this.stack.back();
-      } else {
-        let { rejectOnInvalidatedSave, approval } = this.config;
-        if ( rejectOnInvalidatedSave && approval ) {
-          let rejectedApproval = approval.clone();
-          rejectedApproval.status = this.ApprovalStatus.REJECTED;
-          rejectedApproval.memo = 'Outdated Approval.';
-          this.approvalRequestDAO.put(rejectedApproval).then(o => {
+      if ( this.config.rejectOnInvalidatedSave ) {
+        onSaveRejectOnInvalidated_(isValid, ucj);
+        return;
+      }
+
+      this.notify(this.SUCCESS_UPDATED, '', this.LogLevel.INFO, true);
+      this.stack.back();
+    },
+
+    async function onSaveRejectOnInvalidated_(isValid, ucj) {
+      if ( ! isValid || ucj.status === this.CapabilityJunctionStatus.ACTION_REQUIRED ) {
+        if ( this.config.approval ) {
+          try {
+            const rejectedApproval = approval.clone();
+            rejectedApproval.status = this.ApprovalStatus.REJECTED;
+            rejectedApproval.memo = 'Outdated Approval.';
+            await this.approvalRequestDAO.put(rejectedApproval);
             this.approvalRequestDAO.cmd(foam.dao.DAO.RESET_CMD);
             this.tableViewApprovalRequestDAO.cmd(foam.dao.DAO.RESET_CMD);
             this.approvalRequestDAO.cmd(foam.dao.DAO.PURGE_CMD);
             this.tableViewApprovalRequestDAO.cmd(foam.dao.DAO.PURGE_CMD);
 
-            this.notify(this.SUCCESS_REMOVED, '', this.LogLevel.INFO, true);
+            this.notify(this.SUCCESS_UPDATED, '', this.LogLevel.INFO, true);
             this.pushMenu('approvals', true);
-          }, e => {
+          } catch (e) {
             this.notify(e.message, '', this.LogLevel.ERROR, true);
-          });
+          }
         } else {
-          this.notify(this.SUCCESS_REMOVED, '', this.LogLevel.INFO, true);
```

Comment:
I realize I was wrong about inverting the condition - if you flip the `if` and `else` here then you can use the guard clause approach

```javascript
if ( isValid && ACTION_REQUIRED && this.config.approval ) {
  this.notify(...);
  this.stack.back();
  return;
}

// the rest of the code
```

Reply:
Shouldn't it be if ( isValid && not ACTION_REQUIRED || this.config.approval ) ?